shell/shared/shell.js

```diff
@@ -445,8 +450,78 @@ if (Meteor.isClient) {
   Template.homeLink.helpers({
     origin: getOrigin
   });
-}
 
+  Template.notifications.helpers({
+    notifications: function () {
+      return Notifications.find({userId: Meteor.userId()}, {sort: {timestamp: -1}}).map(function (row) {
+        // TODO(someday): make this work with grains that are not owned by the current user.
+        // Currently we only subscribe to grains owned by this user, and the following query will
```

Comment:
It's a bit worse than that: We only subscribe to all of the user's grain titles when they are looking at the grain list. If they are looking at a specific grain, then we'll only have subscribed to that one grain, and if they're on some other page, we won't have any titles. Currently if a grain is running in the background and you open another grain, the notifications view looks wrong because the grain title is missing.